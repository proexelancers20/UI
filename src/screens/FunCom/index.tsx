import React, {useState} from 'react';
import {
  View,
  Text,
  TextInput,
  Button,
  StyleSheet,
  Pressable,
} from 'react-native';

const MyForm: React.FC = () => {
  const [name, setName] = useState<string>('');
  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');

  const handleNameChange = (text: string) => {
    setName(text);
  };

  const handleEmailChange = (text: string) => {
    setEmail(text);
  };

  const handlePasswordChange = (text: string) => {
    setPassword(text);
  };

  const handleSubmit = () => {
    // Here, you can perform actions like sending data to a server or local storage.
    console.log('Name:', name);
    console.log('Email:', email);
    console.log('Password:', password);

    // Reset the form after submission
    setName('');
    setEmail('');
    setPassword('');
  };

  const Press_Fun = () => {
    return (
      <Pressable onPress={() => {setPassword(`text`);}}>
        <Text>Pressable</Text>
      </Pressable>
    );
  };

  return (
    <View style={styles.container}>
      <Text style={styles.label}>Name:</Text>
      <TextInput
        testID="Enter_Name"
        style={styles.input}
        value={name}
        onChangeText={handleNameChange}
        placeholder="Enter your name"
      />

      <Text style={styles.label}>Email:</Text>
      <TextInput
        testID="Enter_Email"
        style={styles.input}
        value={email}
        onChangeText={handleEmailChange}
        placeholder="Enter your email"
        keyboardType="email-address"
      />

      <Text style={styles.label}>Password:</Text>
      <TextInput
        testID="Enter_Password"
        style={styles.input}
        value={password}
        onChangeText={handlePasswordChange}
        placeholder="Enter your password"
        secureTextEntry={true}
      />

      <Press_Fun />

      <Button title="Submit" onPress={handleSubmit} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
  },
  label: {
    fontSize: 18,
    marginBottom: 8,
  },
  input: {
    fontSize: 16,
    borderWidth: 1,
    borderColor: '#ccc',
    padding: 8,
    marginBottom: 16,
  },
});

export default MyForm;
