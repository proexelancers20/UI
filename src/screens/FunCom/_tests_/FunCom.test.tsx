import React from 'react';
import FunCom from '..';
import {fireEvent, render} from '@testing-library/react-native';

const compoent = <FunCom />;

describe('Function component screen', () => {
  it('snap of the screen', () => {
    const container = render(compoent);
    expect(container.toJSON()).toMatchSnapshot();

    const { getByTestId, getByText } = render(compoent);
    expect(getByText('Name:')).toBeDefined();
    fireEvent.press(getByText('Submit'));
    fireEvent.press(getByText('Pressable'));
    fireEvent.changeText(getByTestId('Enter_Name'));
    fireEvent.changeText(getByTestId('Enter_Email'))
    fireEvent.changeText(getByTestId('Enter_Password'))
  });
});
