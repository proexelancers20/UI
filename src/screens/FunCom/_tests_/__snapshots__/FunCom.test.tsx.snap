// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Function component screen snap of the screen 1`] = `
<View
  style={
    {
      "flex": 1,
      "padding": 16,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 18,
        "marginBottom": 8,
      }
    }
  >
    Name:
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter your name"
    style={
      {
        "borderColor": "#ccc",
        "borderWidth": 1,
        "fontSize": 16,
        "marginBottom": 16,
        "padding": 8,
      }
    }
    testID="Enter_Name"
    value=""
  />
  <Text
    style={
      {
        "fontSize": 18,
        "marginBottom": 8,
      }
    }
  >
    Email:
  </Text>
  <TextInput
    keyboardType="email-address"
    onChangeText={[Function]}
    placeholder="Enter your email"
    style={
      {
        "borderColor": "#ccc",
        "borderWidth": 1,
        "fontSize": 16,
        "marginBottom": 16,
        "padding": 8,
      }
    }
    testID="Enter_Email"
    value=""
  />
  <Text
    style={
      {
        "fontSize": 18,
        "marginBottom": 8,
      }
    }
  >
    Password:
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter your password"
    secureTextEntry={true}
    style={
      {
        "borderColor": "#ccc",
        "borderWidth": 1,
        "fontSize": 16,
        "marginBottom": 16,
        "padding": 8,
      }
    }
    testID="Enter_Password"
    value=""
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text>
      Pressable
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
